addon/components/frost-autocomplete.js

```diff
@@ -0,0 +1,325 @@
+    // dropdown so it can position itself properly relative to the select
+    this.set('_element', this.$())
+  },
+
+  // == DOM Events ============================================================
+
+  _onClick: on('click', function (e) {
+    if (!this.get('disabled')) {
+      this.toggleProperty('opened')
+    }
+    if (this.onClick) {
+      this.onClick(e)
+    }
+  }),
+
+  _onKeyDown: on('keyDown', function (e) {
+    if ([DOWN_ARROW, UP_ARROW].indexOf(e.keyCode) !== -1) {
+      e.preventDefault() // Keep up/down arrow from scrolling page
+      e.stopPropagation()
+      this.set('opened', true)
+    }
+  }),
+
+  _onFocusIn: on('focusIn', function () {
+    // If select is disabled make sure it can't get focus
+    if (this.get('disabled')) {
+      this.$().blur()
+    } else if (!this.get('focused')) {
+      this.set('focused', true)
+
+      if (typeOf(this.onFocus) === 'function') {
+        this.onFocus()
+      }
+    }
+  }),
+
+  _onFocusOut: on('focusOut', function () {
+    if (this.isDestroyed || this.isDestroying) {
+      return
+    }
+
+    this.setProperties({
+      focused: false,
+      opened: false
+    })
+
+    if (typeOf(this.onBlur) === 'function') {
+      this.onBlur()
+    }
+  }),
+
+  /**
+   * Fires input event after waiting for debounceInterval to clear
+   * @param {Function} cb - Reference to onInput
+   * @param {String} value - Filter String
+   */
+  inputTask: task(function * (cb, value) {
+    const debounceInterval = this.get('debounceInterval')
+
+    yield timeout(debounceInterval)
+    cb(value)
+  }).restartable(),
+
+  // == Actions ============================================================
+
+  actions: {
+    closeDropDown () {
+      this.setProperties({
+        filter: '',
+        opened: false
+      })
+    },
+
+    onKeyDown (event) {
+      if ([DOWN_ARROW, UP_ARROW].indexOf(event.keyCode) !== -1) {
+        this._handleArrowKey(event.keyCode === UP_ARROW)
+      } else if (BACKSPACE === event.keyCode) {
+        this.set('_userInput', true)
+      }
+    },
+
+    onKeyPress (event) {
+      if (event.keyCode !== ENTER) {
+        this.setProperties({
+          _userInput: true,
+          opened: true
+        })
+      }
+    },
+
+    selectItem (selectedItem) {
+      this.setProperties({
+        _userInput: false,
+        filter: get(selectedItem, 'label'),
+        internalSelectedItem: selectedItem
+      })
+
+      const onChange = this.get('onChange')
+
+      if (typeOf(onChange) === 'function') {
+        run.next(() => {
+          onChange(get(selectedItem, 'value'))
+        })
+      }
+
+      this._element.find('input').first().focus()
+    },
+
+    onClear () {
+      this.setProperties({
+        _userInput: false,
+        opened: false,
+        internalSelectedItem: undefined
+      })
+
+      const onClear = this.get('onClear')
+
+      if (typeOf(onClear) === 'function') {
+        run.next(() => {
```

Comment:
`run.next` is essentially just a `setTimeout`. You don't gain any protection if the component were destroyed. Either schedule it in a queue or protect it against the component being destroyed.